<?xml version="1.0" encoding="ascii"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
          "DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
  <title>pylogeny.alignment.phylipFriendlyAlignment</title>
  <link rel="stylesheet" href="epydoc.css" type="text/css" />
  <script type="text/javascript" src="epydoc.js"></script>
</head>

<body bgcolor="white" text="black" link="blue" vlink="#204080"
      alink="#204080">
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="pylogeny-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://AlexSafatli.github.io/Pylogeny">Pylogeny</a></th>
          </tr></table></th>
  </tr>
</table>
<table width="100%" cellpadding="0" cellspacing="0">
  <tr valign="top">
    <td width="100%">
      <span class="breadcrumbs">
        <a href="pylogeny-module.html">Package&nbsp;pylogeny</a> ::
        <a href="pylogeny.alignment-module.html">Module&nbsp;alignment</a> ::
        Class&nbsp;phylipFriendlyAlignment
      </span>
    </td>
    <td>
      <table cellpadding="0" cellspacing="0">
        <!-- hide/show private -->
        <tr><td align="right"><span class="options">[<a href="javascript:void(0);" class="privatelink"
    onclick="toggle_private();">hide&nbsp;private</a>]</span></td></tr>
        <tr><td align="right"><span class="options"
            >[<a href="frames.html" target="_top">frames</a
            >]&nbsp;|&nbsp;<a href="pylogeny.alignment.phylipFriendlyAlignment-class.html"
            target="_top">no&nbsp;frames</a>]</span></td></tr>
      </table>
    </td>
  </tr>
</table>
<!-- ==================== CLASS DESCRIPTION ==================== -->
<h1 class="epydoc">Class phylipFriendlyAlignment</h1><p class="nomargin-top"><span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment">source&nbsp;code</a></span></p>
<center>
<center>  <map id="uml_class_diagram_for_pylogeny_3" name="uml_class_diagram_for_pylogeny_3">
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#__init__" title="Instantiate an object intended to wrap an alignment for the purposes of &#160;running phylogenetic inference." alt="" coords="17,377,271,396"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#getPhylip" title="Get a path to a temporary Phylip file." alt="" coords="17,396,271,415"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#writeProperNexus" title="Write a Nexus file with proper names." alt="" coords="17,415,271,433"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#reassignFromReinterpretedNewick" title="Replace all proper names with reassigned names in a Newick tree." alt="" coords="17,433,271,452"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#reinterpretNewick" title="Replaces all reassigned names to proper names in a Newick tree." alt="" coords="17,452,271,471"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#getProperName" title="Return the actual name for an integer&#45;based sequence name that was &#160;reassigned at initialization." alt="" coords="17,471,271,489"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#getTaxa" title="Return current taxa names in the alignment." alt="" coords="17,489,271,508"/>
<area shape="rect" id="node6" href="pylogeny.alignment.phylipFriendlyAlignment-class.html#recreateObject" title="Reintializes the object." alt="" coords="17,508,271,527"/>
<area shape="rect" id="node1" href="pylogeny.alignment.phylipFriendlyAlignment-class.html" title="An alignment object that renames all comprising taxa in order to be able to be written as a Phylip file." alt="" coords="5,345,283,533"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#__getitem__" title="pylogeny.alignment.alignment.__getitem__" alt="" coords="65,39,223,57"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#__str__" title="str(x)" alt="" coords="65,57,223,76"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#__len__" title="pylogeny.alignment.alignment.__len__" alt="" coords="65,76,223,95"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#__iter__" title="pylogeny.alignment.alignment.__iter__" alt="" coords="65,95,223,113"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#close" title="Delete all temporary files and clear data." alt="" coords="65,113,223,132"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#toStrList" title="Get all sequences as a list of strings." alt="" coords="65,132,223,151"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getStateModel" title="pylogeny.alignment.alignment.getStateModel" alt="" coords="65,151,223,169"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getSize" title="Return the size of the alignment, or how many characters there are in each &#160;respective item in the alignment." alt="" coords="65,169,223,188"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getNumSeqs" title="Return the number of sequences that are present in the sequence alignment." alt="" coords="65,188,223,207"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getDim" title="Return the dimensionality of the sequence alignment (how many different &#160;types of characters)." alt="" coords="65,207,223,225"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getSequence" title="Acquire the ith sequence." alt="" coords="65,225,223,244"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getFASTA" title="Get (and create if not already) a path to a temporary FASTA file." alt="" coords="65,244,223,263"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#getApproxMLNewick" title="Get a tree in newick format via use of FastTree that serves as an &#160;approximation of the maximum likelihood tree for this data." alt="" coords="65,263,223,281"/>
<area shape="rect" id="node7" href="pylogeny.alignment.alignment-class.html#bootstrap" title="Perform bootstrapping on the alignment data." alt="" coords="65,300,223,319"/>
<area shape="rect" id="node2" href="pylogeny.alignment.alignment-class.html" title="Wrap a biological sequence alignment to enable functionality necessary for phylogenetic inference." alt="" coords="53,6,235,325"/>
</map>
  <img src="uml_class_diagram_for_pylogeny_3.gif" alt='' usemap="#uml_class_diagram_for_pylogeny_3" ismap="ismap" class="graph-without-title" />
</center>
</center>
<hr />
<p>An alignment object that renames all comprising taxa in order to be 
  able to be written as a Phylip file.</p>

<!-- ==================== INSTANCE METHODS ==================== -->
<a name="section-InstanceMethods"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Instance Methods</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-InstanceMethods"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="pylogeny.alignment.phylipFriendlyAlignment-class.html#__init__" class="summary-sig-name">__init__</a>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">inal</span>=<span class="summary-sig-default">None</span>)</span><br />
      Instantiate an object intended to wrap an alignment for the purposes 
      of running phylogenetic inference.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.__init__">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_makePhylipFile"></a><span class="summary-sig-name">_makePhylipFile</span>(<span class="summary-sig-arg">self</span>)</span><br />
      PRIVATE: Make a temporary Phylip file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment._makePhylipFile">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_makeProperNexusFile"></a><span class="summary-sig-name">_makeProperNexusFile</span>(<span class="summary-sig-arg">self</span>)</span><br />
      PRIVATE: Make a temporary Nexus file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment._makeProperNexusFile">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr class="private">
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="_reassignNames"></a><span class="summary-sig-name">_reassignNames</span>(<span class="summary-sig-arg">self</span>)</span><br />
      PRIVATE: Reassign all sequence names to shorter Phylip-friendly 
      integer-based names.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment._reassignNames">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="pylogeny.alignment.phylipFriendlyAlignment-class.html#getPhylip" class="summary-sig-name">getPhylip</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Get a path to a temporary Phylip file.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.getPhylip">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="writeProperNexus"></a><span class="summary-sig-name">writeProperNexus</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">wri</span>)</span><br />
      Write a Nexus file with proper names.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.writeProperNexus">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="reassignFromReinterpretedNewick"></a><span class="summary-sig-name">reassignFromReinterpretedNewick</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">tr</span>)</span><br />
      Replace all proper names with reassigned names in a Newick tree.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.reassignFromReinterpretedNewick">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="reinterpretNewick"></a><span class="summary-sig-name">reinterpretNewick</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">tr</span>)</span><br />
      Replaces all reassigned names to proper names in a Newick tree.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.reinterpretNewick">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="getProperName"></a><span class="summary-sig-name">getProperName</span>(<span class="summary-sig-arg">self</span>,
        <span class="summary-sig-arg">n</span>)</span><br />
      Return the actual name for an integer-based sequence name that was 
      reassigned at initialization.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.getProperName">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a href="pylogeny.alignment.phylipFriendlyAlignment-class.html#getTaxa" class="summary-sig-name">getTaxa</a>(<span class="summary-sig-arg">self</span>)</span><br />
      Return current taxa names in the alignment.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.getTaxa">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
<tr>
    <td width="15%" align="right" valign="top" class="summary">
      <span class="summary-type">&nbsp;</span>
    </td><td class="summary">
      <table width="100%" cellpadding="0" cellspacing="0" border="0">
        <tr>
          <td><span class="summary-sig"><a name="recreateObject"></a><span class="summary-sig-name">recreateObject</span>(<span class="summary-sig-arg">self</span>)</span><br />
      Reintializes the object.</td>
          <td align="right" valign="top">
            <span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.recreateObject">source&nbsp;code</a></span>
            
          </td>
        </tr>
      </table>
      
    </td>
  </tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="pylogeny.alignment.alignment-class.html">alignment</a></code></b>:
      <code><a href="pylogeny.alignment.alignment-class.html#__getitem__">__getitem__</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#__iter__">__iter__</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#__len__">__len__</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#__str__">__str__</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#bootstrap">bootstrap</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#close">close</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getAlignment">getAlignment</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getApproxMLNewick">getApproxMLNewick</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getApproxMLTree">getApproxMLTree</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getDim">getDim</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getFASTA">getFASTA</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getNumSeqs">getNumSeqs</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getSequence">getSequence</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getSize">getSize</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#getStateModel">getStateModel</a></code>,
      <code><a href="pylogeny.alignment.alignment-class.html#toStrList">toStrList</a></code>
      </p>
    <div class="private">    <p class="indent-wrapped-lines"><b>Inherited from <code><a href="pylogeny.alignment.alignment-class.html">alignment</a></code></b> (private):
      <code><a href="pylogeny.alignment.alignment-class.html#_makeFASTAFile" onclick="show_private();">_makeFASTAFile</a></code>
      </p></div>
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__delattr__</code>,
      <code>__format__</code>,
      <code>__getattribute__</code>,
      <code>__hash__</code>,
      <code>__new__</code>,
      <code>__reduce__</code>,
      <code>__reduce_ex__</code>,
      <code>__repr__</code>,
      <code>__setattr__</code>,
      <code>__sizeof__</code>,
      <code>__subclasshook__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== PROPERTIES ==================== -->
<a name="section-Properties"></a>
<table class="summary" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Properties</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-Properties"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
  <tr>
    <td colspan="2" class="summary">
    <p class="indent-wrapped-lines"><b>Inherited from <code>object</code></b>:
      <code>__class__</code>
      </p>
    </td>
  </tr>
</table>
<!-- ==================== METHOD DETAILS ==================== -->
<a name="section-MethodDetails"></a>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr bgcolor="#70b0f0" class="table-header">
  <td colspan="2" class="table-header">
    <table border="0" cellpadding="0" cellspacing="0" width="100%">
      <tr valign="top">
        <td align="left"><span class="table-header">Method Details</span></td>
        <td align="right" valign="top"
         ><span class="options">[<a href="#section-MethodDetails"
         class="privatelink" onclick="toggle_private();"
         >hide private</a>]</span></td>
      </tr>
    </table>
  </td>
</tr>
</table>
<a name="__init__"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">__init__</span>(<span class="sig-arg">self</span>,
        <span class="sig-arg">inal</span>=<span class="sig-default">None</span>)</span>
    <br /><em class="fname">(Constructor)</em>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.__init__">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Instantiate an object intended to wrap an alignment for the purposes 
  of running phylogenetic inference.</p>
  <dl class="fields">
    <dt>Overrides:
        object.__init__
        <dd><em class="note">(inherited documentation)</em></dd>
    </dt>
  </dl>
</td></tr></table>
</div>
<a name="getPhylip"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">getPhylip</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.getPhylip">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Get a path to a temporary Phylip file. This will be deleted upon 
  closure of the alignment instance.</p>
  <dl class="fields">
  </dl>
</td></tr></table>
</div>
<a name="getTaxa"></a>
<div>
<table class="details" border="1" cellpadding="3"
       cellspacing="0" width="100%" bgcolor="white">
<tr><td>
  <table width="100%" cellpadding="0" cellspacing="0" border="0">
  <tr valign="top"><td>
  <h3 class="epydoc"><span class="sig"><span class="sig-name">getTaxa</span>(<span class="sig-arg">self</span>)</span>
  </h3>
  </td><td align="right" valign="top"
    ><span class="codelink"><a href="pylogeny.alignment-pysrc.html#phylipFriendlyAlignment.getTaxa">source&nbsp;code</a></span>&nbsp;
    </td>
  </tr></table>
  
  <p>Return current taxa names in the alignment.</p>
  <dl class="fields">
    <dt>Overrides:
        <a href="pylogeny.alignment.alignment-class.html#getTaxa">alignment.getTaxa</a>
    </dt>
  </dl>
</td></tr></table>
</div>
<br />
<!-- ==================== NAVIGATION BAR ==================== -->
<table class="navbar" border="0" width="100%" cellpadding="0"
       bgcolor="#a0c0ff" cellspacing="0">
  <tr valign="middle">
  <!-- Home link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="pylogeny-module.html">Home</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Tree link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="module-tree.html">Trees</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Index link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="identifier-index.html">Indices</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Help link -->
      <th>&nbsp;&nbsp;&nbsp;<a
        href="help.html">Help</a>&nbsp;&nbsp;&nbsp;</th>

  <!-- Project homepage -->
      <th class="navbar" align="right" width="100%">
        <table border="0" cellpadding="0" cellspacing="0">
          <tr><th class="navbar" align="center"
            ><a class="navbar" target="_top" href="http://AlexSafatli.github.io/Pylogeny">Pylogeny</a></th>
          </tr></table></th>
  </tr>
</table>
<table border="0" cellpadding="0" cellspacing="0" width="100%%">
  <tr>
    <td align="left" class="footer">
    Generated by Epydoc 3.0.1 on Thu Nov 20 14:00:41 2014
    </td>
    <td align="right" class="footer">
      <a target="mainFrame" href="http://epydoc.sourceforge.net"
        >http://epydoc.sourceforge.net</a>
    </td>
  </tr>
</table>

<script type="text/javascript">
  <!--
  // Private objects are initially displayed (because if
  // javascript is turned off then we want them to be
  // visible); but by default, we want to hide them.  So hide
  // them unless we have a cookie that says to show them.
  checkCookie();
  // -->
</script>
</body>
</html>
